cmd/govim/signs.go

```diff
@@ -27,13 +27,40 @@ type defineDict struct {
 
 // signDefine defines the sign types (sign names) and must be called once before placing any signs
 func (v *vimstate) signDefine() error {
-	for _, hi := range []config.Highlight{config.HighlightSignErr, config.HighlightSignWarn, config.HighlightSignInfo, config.HighlightSignHint} {
+	signnames := []config.Highlight{
```

Comment:
Not sure this will work if the user makes a call to:

```
call sign_define("GOVIMSignErr",{"text":"!!","texthl":"GOVIMSignErr"})
```

after startup? Arguably that will happen less often, but we should cover that case too.

i.e. we need to hook something into `*vimstate.setConfig` to have this working 100%

Reply:
It does actually work already, re-defining an existing sign will overwrite, so that the new one is used instead.

Reply:
Yes but you need something like an edit after changing the sign to see it take effect. Hence I think we need a hook in `*vimstate.setConfig` to trigger replacing of the signs, no?

Reply:
Indeed, moving the cursor is enough to see it take effect. Is it possible for `govim` to detect such call? Since it isn't a part of `govim`'s configuration at all (it's all handled entirely in the context of vim).

I could raise an issue towards vim, like:
>  Existing signs isn't updated after a `sign_define()` call

Reply:
> Indeed, moving the cursor is enough to see it take effect.

Is it? In my tests locally a change was required in the file itself. Perhaps my test was bad?

Reply:
Hm, yeah. At least in Linux using Vim 8.2.0032 via mosh and macOS using Vim 8.2 (0000).

I open up a file in a temp directory (with go.mod):
```
package main

func main() {
	foo
}
```
Then call:
```
:call sign_define("GOVIMSignErr", {"text":"aa","texthl":"GOVIMSignErr"})
```
And move the cursor one step with the arrow keys.

Reply:
That worked! No idea what I did before. 

Still, I think we should fix it so that the user isn't left wondering why nothing changed on screen, would you agree?

Reply:
Yeah, agree fully. I’m not sure however if it’s possible to hook on to a ordinary call in vim. I haven’t seen that it generates any event, as it is handled outside of govim entirely.

We could create a govim config that “should” be used instead, but I don’t think that this scenario is common enough so that it is worth it. A user would still be able to call sign_define and run into the same issue.

Any other ideas how we could hook on to sign_define? Is it possible to override & wrap the ordinary one with a refresh? I’ll give it a try later tonight unless you already know.

Reply:
Good points all round :)

This actually smells more like a Vim issue than one for us.

Let's leave it as you have it for now on the `govim` side and then raise an issue with `vim_dev`?

Reply:
:+1: I rebased this PR against latest master & wrote https://github.com/vim/vim/issues/5455